### DISCUSSION TITLE
Fill Linear Gradient color VictoryBar typescript project

### DISCUSSION THREAD

**Discussion by kieuquang1110tt** [FormidableLabs/victory#2118] [Q&A]
I try to use fill color bar but nothing work. Help.

import React, { memo } from "react";
import { View, Image } from "react-native";
import {
  TopNavigation,
  useTheme,
  StyleService,
  useStyleSheet,
  Layout,
} from "@ui-kitten/components";
import { useNavigation } from "@react-navigation/native";
import { useSafeAreaInsets } from "react-native-safe-area-context";
import useLayout from "hooks/useLayout";
import { globalStyle } from "styles/globalStyle";

import Text from "components/Text";
import Content from "components/Content";
import Container from "components/Container";
import {
  VictoryBar,
  VictoryGroup,
  VictoryLabel,
  VictoryChart,
  VictoryTooltip,
  VictoryVoronoiContainer,
  VictoryArea,
} from "victory-native";
import Svg, { Defs, LinearGradient, Stop } from "react-native-svg";
import { url } from "inspector";
interface Props {
  x: number;
  y: number;
  y0: number;
}
interface DataProps {
  data: Props[];
  setStep: number;
}

const ActivityChart = memo(({ data, setStep }: DataProps) => {
  const { goBack } = useNavigation();
  const { height, width, top, bottom } = useLayout();
  const theme = useTheme();
  const styles = useStyleSheet(themedStyles);
  return (
    <Layout style={styles.container}>
      <View>
        <View>
          <Svg style={{ height: 0, width: 0 }} key={"gradient"}>
            <Defs key={"gradient"}>
              <LinearGradient id="gradient" x1={"10"} x2={"100%"} y2={"10%"}>
                <Stop offset="0%" stopColor="#fb4867" />
                <Stop offset="25%" stopColor="#fd8153" />
                <Stop offset="50%" stopColor="#fbb642" />
                <Stop offset="75%" stopColor="#60cb7e" />
                <Stop offset="100%" stopColor="#1bd499" />
              </LinearGradient>
            </Defs>
          </Svg>
        </View>
        <VictoryBar
          data={data}
          cornerRadius={{ bottom: 4, top: 4 }}
          barWidth={6}
          animate={{
            duration: 2000,
            onLoad: { duration: 1000 },
          }}
          height={260}
          style={{
            data: {
              fill: "url(#gradient)",
              // fill: "#1E93FA",
              stroke: "#1E93FA",
              strokeWidth: 2,
            },
          }}
        ></VictoryBar>
      </View>
    </Layout>
  );
});

export default ActivityChart;

const themedStyles = StyleService.create({
  container: {
    flex: 1,
  },
});

**Comment by becca-bailey** [comment]
Hi @kieuquang1110tt, when you get a chance do you mind throwing this in code sandbox? That would make it easier for me to see what is going on.

My guess at the moment is that the Victory `style` prop only supports specific attributes, and you might need to use a custom component to have a little more control.

```jsx
<VictoryBar dataComponent={<Bar pathComponent={<path fill="url(#gradient") />} />} />
```

**Comment by gonganruyi** [comment]
react native 中如何实现？？

**Comment by gonganruyi** [comment]
@kieuquang1110tt 可以实现，采用以下代码：

```
 <VictoryChart padding={0} width={146}
    height={58}>
    <Defs>
      <LinearGradient id="charGradient" x1="0.5" y1="0" x2="0.5" y2="1">
        <Stop offset="0" stopColor={chartColor} />
        <Stop offset="1" stopColor="#FFFFFF" />
      </LinearGradient>
    </Defs>
    <VictoryArea
      interpolation="basis"
      width={146}
      height={58}
      padding={0}
      data={chartData}
      style={{
        data: { fill: 'url(#charGradient)', fillOpacity: 0.3, stroke: rateColor, strokeWidth: 1 },
        parent: { paddingTop: 0 }
      }}
    />
    <VictoryAxis style={{
      axis: { stroke: "#fff" },
      grid: { strokeOpacity: 0 },
      ticks: { stroke: "grey", size: 0 },
    }} />
 </VictoryChart>
```

**Comment by becca-bailey** [comment]
Hi @kieuquang1110tt, I'm going to convert this help request to a discussion thread to prevent it from getting lost in issue reports. Thanks!
